# CSV súbor

In [ ]:
csv_file = '/Users/educa/Documents/STUDY/GOPAS/subory/DATA/OBCE_GPS.csv'

In [2]:
import pandas as pd

pandas funkcia read_csv() načíta dáta zo súboru do dataframu

In [3]:
data = pd.read_csv(csv_file)
data.__class__

pandas.core.frame.DataFrame

In [4]:
data.head()

Unnamed: 0        DOBEC         OBEC            OKRES     PSC      DPOSTA  \
0           0      Ábelová      Ábelová          Lučenec  985 13     Ábelová   
1           1       Abovce       Abovce  Rimavská Sobota  980 44  Lenartovce   
2           2      Abrahám      Abrahám          Galanta  925 45     Abrahám   
3           3  Abrahámovce  Abrahámovce         Bardejov  086 41   Raslavice   
4           4  Abrahámovce  Abrahámovce         Kežmarok  059 72       Vrbov   

        POSTA  KOD_OKR KRAJ        LAT        LNG  
0     Ábelová      606   BC  48.410817  19.431741  
1  Lenartovce      609   BC  48.318828  20.339351  
2     Abrahám      202   TA  48.250846  17.616963  
3   Raslavice      701   PV  49.161432  21.343484  
4       Vrbov      703   PV  49.044580  20.439083

In [5]:
data.shape

(4210, 11)

In [6]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4210 entries, 0 to 4209
Data columns (total 11 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  4210 non-null   int64  
 1   DOBEC       4210 non-null   object 
 2   OBEC        4210 non-null   object 
 3   OKRES       4210 non-null   object 
 4   PSC         4154 non-null   object 
 5   DPOSTA      4154 non-null   object 
 6   POSTA       4210 non-null   object 
 7   KOD_OKR     4210 non-null   int64  
 8   KRAJ        4210 non-null   object 
 9   LAT         4210 non-null   float64
 10  LNG         4210 non-null   float64
dtypes: float64(2), int64(2), object(7)
memory usage: 361.9+ KB


Dataframe obsahuje 4210 riadkov a 11 stĺpcov. Názvy stĺpcov a typy - viď vyššie

In [7]:
data.__sizeof__()/1024/1024 #megabyte

2.05999755859375

pandas ma extremne velke systemovane poziadavky na dataframy. Na disku má tento súbor 416 kB

In [8]:
novy_subor = '/Users/educa/Documents/STUDY/GOPAS/subory/DATA/OBCE_GPS_new.csv'
data.to_csv(novy_subor, index=False)
# parameter index=False 